src/rules/spaceWithinParensRule.ts

```diff
@@ -0,0 +1,164 @@
+        `,
+        options: { type: "number", min: 0 },
+        type: "style",
+        typescriptOnly: false,
+    };
+    /* tslint:enable:object-literal-sort-keys */
+
+    public static FAILURE_NO_SPACE = "No whitespace within parentheses";
+    public static FAILURE_NEEDS_SPACE = "Needs whitespace within parentheses";
+    public static FAILURE_NO_EXTRA_SPACE(count: number): string {
+        return `No more than ${count} whitespace within parentheses allowed`;
+    }
+
+    public apply(sourceFile: ts.SourceFile): Lint.RuleFailure[] {
+        return this.applyWithWalker(new SpaceWhitinParensWalker(sourceFile, this.ruleName, parseOptions(Number(this.ruleArguments[0]))));
+    }
+}
+
+function parseOptions(whitespaceSize?: number): Options {
+    let size = 0;
+    if (typeof whitespaceSize === "number") {
+        if (whitespaceSize >= 0) {
+            size = whitespaceSize;
+        }
+    } else if (whitespaceSize !== undefined) {
+        const parsedSize = parseInt(whitespaceSize, 10);
+        if (!Number.isNaN(parsedSize) && parsedSize >= 0) {
+            size = parsedSize;
+        }
+    }
+    return {
+        size,
+    };
+}
+
+class SpaceWhitinParensWalker extends Lint.AbstractWalker<Options> {
+    public walk(sourceFile: ts.SourceFile) {
+        return ts.forEachChild(sourceFile, (node: ts.Node): void => {
+            this.checkNode(node);
+        });
+    }
+
+    private checkNode(node: ts.Node): void {
+        forEachToken(node, (token: ts.Node) => {
+            if (token.kind === ts.SyntaxKind.OpenParenToken) {
+                this.checkOpenParenToken(token);
+            } else if (token.kind === ts.SyntaxKind.CloseParenToken) {
+                this.checkCloseParenToken(token);
+            }
+        });
+    }
+
+    private checkOpenParenToken(tokenNode: ts.Node) {
+        let currentPos = tokenNode.end;
+        let currentChar = this.sourceFile.text.charCodeAt(currentPos);
+        const allowedSpaceCount = this.options.size;
+
+        while (ts.isWhiteSpaceSingleLine(currentChar)) {
+            ++currentPos;
+            currentChar = this.sourceFile.text.charCodeAt(currentPos);
+        }
+        if (!ts.isLineBreak(currentChar)) {
+            const whitespaceCount = currentPos - tokenNode.end;
+            if (whitespaceCount !== allowedSpaceCount) {
+                let length = 0;
+                let pos = tokenNode.end;
+
+                if (whitespaceCount > allowedSpaceCount) {
+                    pos += allowedSpaceCount;
+                    length = whitespaceCount - allowedSpaceCount;
+                } else if (whitespaceCount > 0 && whitespaceCount < allowedSpaceCount) {
+                    pos += allowedSpaceCount - whitespaceCount;
+                }
+                this.AddFailureAtWithFix(pos, length, whitespaceCount);
+            }
+        }
+    }
+
+    private checkCloseParenToken(tokenNode: ts.Node) {
+        let currentPos = tokenNode.getStart() - 1;
```

Comment:
Can be simplified to:
```ts
let currentPos = tokenNode.end - 2; // close paren token has a known width of 1, therefore node.getStart() === node.end - 1
```

Otherwise I would suggest to pass `sourceFile` as parameter to `getStart` for performance reasons.